### DISCUSSION TITLE
How to modify the response based on POST body params

### DISCUSSION THREAD

**Discussion by pstephenwille** [cypress-io/cypress#15633] [Component Testing]
I have a shared setup file that defines the intercepted routes.  The default response for those routes is provided by JSON-server.  In the individual test files, I override the route and response data that's specific to the test.  This allows me to have a centrally defined route for most cases and only override certain ones needed to test a code path.   The problem I'm having now is, I can't change the response data once the route is `intercepted` in the shared file.

I have a shared default route defined like so:
```
           cy.intercept('POST', new RegExp(CONST.URI.FOO))
            .as('allFoo');
```

And I'd like to override it in the test like so
```
            cy.intercept('POST', new RegExp(CONST.URI.FOO + '/query'), (req) => {
                const foo = req.body.filterAttributes.includes('x');
                req.reply((resp) => {
                    if (!foo) {
                        resp.body.fields[0].updatedAt = getMonthDayYearAndTime(new Date());
                    } else {
                        resp.body = {fields: fixtureData};
                    }
                });
            });
```

Is this even possible?
Perhaps I could only define the route in the tests that use it but that is now how our tests are structured.
Or perhaps a `cy.intercept(url).clear` to allow us to remock the route

**Comment by herecydev** [comment]
Unfortunately not at the moment, see https://github.com/cypress-io/cypress/issues/9302

It however is highly desirable, and PRs are out to make this so: https://github.com/cypress-io/cypress/pull/14543
